playbooks/portals-deploy.yml

```diff
@@ -2,19 +2,27 @@
   hosts: webportals
   gather_facts: False
 
-  # Limit concurrency to keep batch_hosts order deterministic
+  # Limit concurrency to keep batch_hosts order deterministic.
+  # When you want to deploy in parallel, keep 'serial: 1' for this play,
+  # parallel deploys are set by 'serial' variable in the next play.
   serial: 1
 
   # Stop on first error, do not execute on the next host
   any_errors_fatal: True
 
   vars:
+    max_hosts: "{{ groups['webportals'] | length - 1 }}"
+
     # Default batch config (one batch with all selected hosts will be created),
     # can be overriden with cli flag --extra-vars/-e
     batch_size: 1
     batch_number: 1
 
   tasks:
+    # Check '--limit' is used
+    - name: Fail if you are targeting all dev and prod webportals
```

Comment:
It only now occurs to me that if we're targeting all prod portals this won't fail. Let's change it to fail when we target all prod portals (or at least as many portals as we have in the prod group).

Reply:
Updated.